src/Routing/Route/DashedRoute.php

```diff
@@ -56,11 +56,12 @@ protected function _camelizePlugin($plugin)
      * camelBacked form.
      *
      * @param string $url The URL to parse
+     * @param string $method The HTTP method.
      * @return array|false An array of request parameters, or false on failure.
      */
-    public function parse($url)
+    public function parse($url, $method = '')
```

Comment:
Our [compatibility guidelines](http://book.cakephp.org/3.0/en/contributing/backwards-compatibility.html#working-on-cakephp) specifically call out adding arguments with default values as :ok: 

I figured the addition of the default argument would be something mentioned in the 3.3 migration guide.